### ISSUE TITLE
All entries missing when change plugins.js file i18n to false

### ISSUE THREAD

**Issue by whidy** [strapi/strapi#13357]
### Required System information

- Node.js version:16.14.2
- NPM version: 8.5.0
- Strapi version: 4.1.11
- Database: SQLite
- Operating system: macOS Monterey (12.3.1)

### Describe the bug

When I try to disable i18n plugin through `plugins.js`, and set `i18n: false`, server get error and restore enable that, all entries missing, include my `new locale` added before missing too.

### Steps to reproduce the behavior

1. Follow [Quick Start Guide](https://docs.strapi.io/developer-docs/latest/getting-started/quick-start.html) to make sure project setup well.
2. let's add `i18n` enable to Restaurant;
![image](https://user-images.githubusercontent.com/2074757/169216456-3bf9cb9f-8c46-423f-9da0-1ff7ed9ac817.png)
3. add a entry in `restaurant` 
![image](https://user-images.githubusercontent.com/2074757/169217749-1303f2e2-e9e2-4243-8b93-b35c63c46286.png)
4. create a file: path: `./config/plugins.js`, and edit text to disable i18n
```js
module.exports = ({ env }) => ({
  i18n: false,
});
```
5. reload admin page will tell us: **Warning: An error occurred**
![image](https://user-images.githubusercontent.com/2074757/169217266-b3ff0060-24c3-42db-8462-6280cef1ad62.png)
6. restore code to open i18n and reload admin panel
```
module.exports = ({ env }) => ({
  // i18n: false,
});
```
8. There is no entries left. 
![image](https://user-images.githubusercontent.com/2074757/169217479-edfc7f45-c32b-4955-9a42-712fe8b4d450.png)

### Expected behavior

When i restore enable i18n, my entries should show as well.

### Screenshots

Screenshots in Steps to reproduce the behavior

### Additional context

Though database has entries, data not lost, but when i18n option changes(restore), admin panel not show entries normally. We can add new entry will see that ID has incresed.

![image](https://user-images.githubusercontent.com/2074757/169219940-d21b47a2-020f-47c4-948a-16840325eb29.png)

BTW, if I set public role has permission find entry, I visited: <http://localhost:1337/api/restaurants> also only show one entry which published.

![image](https://user-images.githubusercontent.com/2074757/169220150-c20d3612-56ab-4fa0-a2ba-1a044a95bd7e.png)

**Comment by whidy** [comment]
Is that plugins.js settings do the same thing as web trigger this warning

![image](https://user-images.githubusercontent.com/2074757/169223576-9042f6bc-ccab-47ce-a40e-9cfd9cc6d486.png)

```
Disabling localization will engender the deletion of all your content but the one associated to your default locale (if existing).
Do you want to disable it?
```

but I don't know this before, it is real very dangerous, but there will happen to others who don't know it.

**Comment by whidy** [comment]
At last, I found a solution.

Because I first found the problem with my local development, my entries miss, I was shocking. So I set a new project to reproduce bug.

The only different thing is database, my project with PostgreSQL.

Now, I found, that `i18n: false` will remove the `locale` column, and true will add colume (but all records value set as `NULL`) . before you change `plugins.js`, locale has value with what you have set in `settings/internationalization`, such as `en`.

![image](https://user-images.githubusercontent.com/2074757/169227935-b26e42d2-39ed-4d98-88ec-14483dd48cf7.png)

But change `plugins.js` only remove/add colume, will not add value.

![image](https://user-images.githubusercontent.com/2074757/169226981-25d23004-5936-4c18-ab75-27f11c760c0b.png)

IF ALL ENTRIES MISSING, TRY ADD VALUE `en` (DEFAULT) TO LOCALE COLUME, THEN TO FIX THE ENTRIES WHICH IS NOT `en`.

EVERY THING GOES WELL. HOPE MY RESEARCH FOR SERVERAL HOURS WOULD HELP YOU : )

**Comment by whidy** [comment]
At last, I found a solution.

Because I first found the problem with my local development, my entries miss, I was shocking. So I set a new project to reproduce bug.

The only different thing is database, my project with PostgreSQL.

Now, I found, that `i18n: false` will remove the `locale` column, and true will add colume (but all records value set as `NULL`) . before you change `plugins.js`, locale has value with what you have set in `settings/internationalization`, such as `en`.

![image](https://user-images.githubusercontent.com/2074757/169227935-b26e42d2-39ed-4d98-88ec-14483dd48cf7.png)

But change `plugins.js` only remove/add colume, will not add value.

![image](https://user-images.githubusercontent.com/2074757/169226981-25d23004-5936-4c18-ab75-27f11c760c0b.png)

IF ALL ENTRIES MISSING, TRY ADD VALUE `en` (DEFAULT) TO LOCALE COLUME, THEN TO FIX THE ENTRIES WHICH IS NOT `en`.

EVERY THING GOES WELL. HOPE MY RESEARCH FOR SERVERAL HOURS WOULD HELP YOU : )
